modules/object-cache/health/load.php

```diff
@@ -0,0 +1,225 @@
+	 */
+	$notes = apply_filter( 'site_status_persistent_object_cache_notes', $notes, $available_services );
+
+	$result['status']       = 'recommended';
+	$result['label']        = __( 'You should use a persistent object cache', 'performance-lab' );
+	$result['description'] .= sprintf(
+		'<p>%s</p>',
+		wp_kses(
+			$notes,
+			array(
+				'a'      => array( 'href' => true ),
+				'code'   => true,
+				'em'     => true,
+				'strong' => true,
+			)
+		)
+	);
+
+	return $result;
+}
+
+/**
+ * Callback for `site_status_persistent_object_cache` filter.
+ *
+ * Determines whether to suggest using a persistent object cache.
+ *
+ * @param mixed $should_suggest Whether to suggest using a persistent object cache.
+ * @return bool
+ */
+function oc_health_should_persistent_object_cache( $should_suggest ) {
+	global $wpdb;
+
+	$thresholds = array(
+		'alloptions_count' => 1,
+		'alloptions_bytes' => 100000,
+		'comments_count'   => 1000,
+		'options_count'    => 1000,
+		'posts_count'      => 1000,
+		'terms_count'      => 1000,
+		'users_count'      => 1000,
+	);
+
+	$alloptions = wp_load_alloptions();
+
+	if ( $thresholds['alloptions_count'] < count( $alloptions ) ) {
+		return true;
+	}
+
+	if ( $thresholds['alloptions_bytes'] < strlen( serialize( wp_load_alloptions() ) ) ) {
+		return true;
+	}
+
+	$table_names = implode( "','", array( $wpdb->comments, $wpdb->options, $wpdb->posts, $wpdb->terms, $wpdb->users ) );
+
+	// With InnoDB the `TABLE_ROWS` are estimates, which are accurate enough and faster to retrieve than individual `COUNT()` queries.
+	$results = $wpdb->get_results(
+		$wpdb->prepare(
+			// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+			"SELECT TABLE_NAME AS 'table', TABLE_ROWS AS 'rows', SUM(data_length + index_length) as 'bytes'
+			FROM information_schema.TABLES
+			WHERE TABLE_SCHEMA = %s
+			AND TABLE_NAME IN ('$table_names')
+			GROUP BY TABLE_NAME;",
+			// phpcs:enable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+			DB_NAME
+		),
+		OBJECT_K
+	);
+
+	if ( $thresholds['comments_count'] < $results[ $wpdb->comments ]->rows ) {
+		return true;
+	}
+
+	if ( $thresholds['options_count'] < $results[ $wpdb->options ]->rows ) {
+		return true;
+	}
+
+	if ( $thresholds['posts_count'] < $results[ $wpdb->posts ]->rows ) {
+		return true;
+	}
+
+	if ( $thresholds['terms_count'] < $results[ $wpdb->terms ]->rows ) {
+		return true;
+	}
+
+	if ( $thresholds['users_count'] < $results[ $wpdb->users ]->rows ) {
+		return true;
+	}
+
+	return false;
+}
+
+/**
+ * Returns a list of available persistent object cache services.
+ *
+ * @return array The list of available persistent object cache services.
+ */
+function oc_health_available_persistent_object_cache_services() {
+	$extensions = array_map(
+		'extension_loaded',
+		array(
+			'Redis'     => 'redis',
+			'Relay'     => 'relay',
+			'Memcached' => 'memcache', // The `memcached` extension seems unmaintained.
+			'APCu'      => 'apcu',
```

Comment:
Memcached, Relay (to some degree), and APCu are **impersistent** in-memory object cache storages. Do we really need to suggest only **persistent** object cache storages?

Reply:
In WordPress `persistent` means longer than the request. Redis can persist data between restarts JFYI.

Reply:
> In WordPress `persistent` means longer than the request.

This is confusing since in non-WordPress world it has a different meaning. Maybe we should remove `persistent` from the function name and use just `oc_health_available_object_cache_services` to avoid confusion? Anyway the filter that is applied to the returned services list doesn't have `persistent` keyword either.

> Redis can persist data between restarts JFYI.

Yes, i know it. That's why Redis considered as persistent object cache storage.